File: src/main/Java/boj_step16/boj2164.java
import java.io.*;
import java.util.LinkedList;
import java.util.Queue;

public class boj2164 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        Queue<Integer> cardSet = new LinkedList<>();
        for (int i = 1; i <= n; i++) cardSet.offer(i);
        while (cardSet.size() != 1) {
            cardSet.poll();
            cardSet.offer(cardSet.poll());
        }
        writer.append(String.valueOf(cardSet.poll())).close();
    }
}

File: src/main/boj_step16/boj2164.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val cardset = LinkedList<Int>()
    repeat(n) {
        cardset.add(it + 1)
    }
    while(cardset.size != 1) {
        cardset.removeFirst()
        cardset.add(cardset.removeFirst())
    }
    w.write(cardset.pop().toString())
    w.close()
}